src/txn/txn_rollback_to_stable.c

```diff
@@ -275,22 +259,24 @@ __txn_rollback_to_stable_btree_walk(
     WT_SESSION_IMPL *session, wt_timestamp_t *rollback_timestamp)
 {
 	WT_DECL_RET;
-	WT_PAGE *page;
 	WT_REF *ref;
 
 	/* Walk the tree, marking commits aborted where appropriate. */
 	ref = NULL;
-	while ((ret = __wt_tree_walk_custom_skip(session, &ref,
-	    __txn_rollback_to_stable_custom_skip,
-	    NULL, WT_READ_NO_EVICT)) == 0 && ref != NULL) {
-		page = ref->page;
+	while ((ret = __wt_tree_walk(session, &ref,
+	    WT_READ_CACHE | WT_READ_LOOKASIDE | WT_READ_NO_EVICT)) == 0 &&
+	    ref != NULL) {
```

Comment:
I don't have all the details of rollback in my head at the moment, but won't this walk no longer visit refs with the `WT_REF_DELETED` state?  It looks like the code tried pretty hard to cover that case previously and AFAICT it no longer does.

I don't know if we have a test that combines `truncate` with a `rollback_to_stable` -- MongoDB's oplog truncates (the only use of truncate in MongoDB) should never be rolled back by reconcilation.

Reply:
@michaelcahill 
>  but won't this walk no longer visit refs with the WT_REF_DELETED state?

I don't think so: the `WT_READ_CACHE` tests in the walk code appear before the `WT_REF_DELETED` checks, so I believe they will be visited.

There's also a question of what the underlying read code itself does, but again I think they will be visited.